samples/nrf9160/modem_shell/src/cloud/cloud_shell.c

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <zephyr.h>
+#include <net/nrf_cloud.h>
+#include <shell/shell.h>
+#include "mosh_print.h"
+
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+#include <net/nrf_cloud_agps.h>
+#endif
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+#include <net/nrf_cloud_pgps.h>
+#endif
+
+BUILD_ASSERT(
+	IS_ENABLED(CONFIG_NRF_CLOUD_MQTT) &&
+	IS_ENABLED(CONFIG_NRF_CLOUD_CONNECTION_POLL_THREAD));
+
+static struct k_work_delayable cloud_reconnect_work;
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+static struct k_work notify_pgps_work;
+#endif
+
+static int cloud_shell_print_usage(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret = 1;
+
+	if (argc > 1) {
+		mosh_error("%s: subcommand not found", argv[1]);
+		ret = -EINVAL;
+	}
+
+	shell_help(shell);
+
+	return ret;
+}
+
+static void cloud_reconnect_work_fn(struct k_work *work)
+{
+	int err = nrf_cloud_connect(NULL);
+
+	if (err == NRF_CLOUD_CONNECT_RES_ERR_ALREADY_CONNECTED) {
```

Comment:
Quickly what we discussed. This could be organized so that first comparison is for NRF_CLOUD_CONNECT_RES_SUCCESS so it would work even if enum values would change (not that it would happen but I think the code is more self-documentary that way).
```
if (err == NRF_CLOUD_CONNECT_RES_SUCCESS) {
	mosh_print("Connection to nRF Cloud re-established");
} else if (err == NRF_CLOUD_CONNECT_RES_ERR_ALREADY_CONNECTED) {
	mosh_print("nRF Cloud connection already established");
} else {
	mosh_error("nrf_cloud_connect, error: %d", err);
}
```

And the same code is in `cmd_cloud_connect()` with very little text changes. If the disconnect event seems to be coming, I think we could consider `cmd_cloud_connect()` to call this function.